mate-panel/panel.c

```diff
@@ -413,17 +413,36 @@ panel_key_press_event (GtkWidget   *widget,
 	return FALSE;
 }
 
+static gboolean
+can_set_background_image (PanelToplevel *toplevel,
```

Comment:
This function is called more than once time, after dropping an image, and after clicking on "Set Background Image" button.

Reply:
Yes, because I left the check in `set_background_image_from_uri()` itself, which is called by the dialog response handler.
I can get the check out of that function and guard the other call, but I reckon the overhead is not a big deal, and anyway the most likely "costly" call is needed in both instances (get a filename from URI).
I it's really a big deal, I can refactor the code not to do this double checking, but OTOH there can be an arbitrary amount of time between the 2 calls (as it waits for user input), so lockdown *could* have been enabled in the meantime which doesn't make this as redundant (unlikely though, I agree).